mitodl/mitxonline
# Implemented enrollment and notification from product detail

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #34

#### What's this PR do?
- Allows users to enroll in course runs from the product detail page.
- Brings feature flag handling up to speed w/ other projects, and adds the `IGNORE_EDX_FAILURES` feature flag, which allows the app to work with enrollments without needing a working edX instance with the correct course runs seeded.
- Adds a mechanism for showing user notifications that originate from outside of the React app context (in this case, the mechanism is used to show a success message when the enrollment succeeds, and it should be applicable for showing a success message when a user first authenticates – #69)

#### How should this be manually tested?
This can be tested with or without an edX instance running with matching course runs in its database. To test _without_ edX, you can set `FEATURE_IGNORE_EDX_FAILURES=True` in your `.env`

1. Set up an enrollable course run. Make sure its parent course has a CMS page, and that you aren't currently enrolled in the run
2. Navigate to the course detail page
3. Click "Enroll"

You should be redirected to the dashboard page, and the "Success!" message in the above screenshot should be shown. On the backend, an API call should have been made to edX to enroll the user in the course with `audit` mode, and a `CourseRunEnrollment` record should be created if the enrollment in edX was successful, or if you have the above feature flag turned on.

After that, visit the course detail page again. The button should say "Enrolled", and if the course run has a courseware URL set, the button should link to that course in edX.

If you're logged out, the "Enrolled" button should redirect you to the login page, which should then redirect you back to the course detail page after you've logged in

#### Screenshots (if appropriate)
<img width="1094" alt="ss 2021-08-13 at 16 50 49 " src="https://user-images.githubusercontent.com/14932219/129422286-ec48d29e-ca64-43a0-a554-673ac2a9e067.png">
<img width="674" alt="ss 2021-08-13 at 16 51 33 " src="https://user-images.githubusercontent.com/14932219/129422296-12b736c2-7287-4c38-abff-d9b7183109da.png">
